### PR TITLE
`XPC` product can embed dynamic libraries and frameworks

### PR DESCRIPTION
**Pull request by PaulTaykalo**
Resolves: No issue was created, but it is impossible to embed dynamic libraries to the XPC, while it is should be alloed 

### Short description 📝

> Describe here the purpose of your PR.

This Pull request allows having dynamic libraries in the `XPC` services.
`XPC` was introduced in https://github.com/tuist/tuist/pull/5077 but for some reason dynamic libraries weren't added as a valid option for XPC https://github.com/tuist/tuist/pull/5077/files#diff-bb33503408b0fd321d9ca4ff9ddd6bb15edcb6bbcbf630daad76711a01169acaR449-R452

### How to test the changes locally 🧐
Run tuist generate for macos_app_with_xpc project

> Include a set of steps for the reviewer to test the changes locally (see [the documentation](https://docs.tuist.io/contributors/testing-strategy) for reference).

One should expect that dynamic libraries are embedded inside XPC

image

### Contributor checklist ✅

- [x] The code has been linted using run `./fourier lint tuist --fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Tests/TuistGeneratorTests/Linter/GraphLinterTests.swift [hunk 1]
```diff
@@ -305,6 +305,51 @@ final class GraphLinterTests: TuistUnitTestCase {
...
     }
 
+    func test_lint_xpcCanDependOnAllTypesOfFrameworksAndLibraries() throws {
+        // Given
+        let path: AbsolutePath = "/project"
+        let dynamicFramework = Target.empty(name: "DynamicFramework", destinations: [.mac], product: .framework)
+        let dynamicLibrary = Target.empty(name: "DynamicLibrary", destinations: [.mac], product: .dynamicLibrary)
+        let staticFramework = Target.empty(name: "StaticFramework", destinations: [.mac], product: .staticFramework)
+        let staticLibrary = Target.empty(name: "StaticLibrary", destinations: [.mac], product: .staticLibrary)
+
```
Comment (daumantas-versockas) [inline Tests/TuistGeneratorTests/Linter/GraphLinterTests.swift:315]:
```suggestion

// Given: Value Graph
```

File: Tests/TuistGeneratorTests/Linter/GraphLinterTests.swift [hunk 2]
```diff
@@ -305,6 +305,52 @@ final class GraphLinterTests: TuistUnitTestCase {
...
+// Given: Value Graph
```
Comment (kwridan) [inline Tests/TuistGeneratorTests/Linter/GraphLinterTests.swift:316]: nit: the indentation looks a bit off :/

**Review by kwridan — APPROVED** [review]
Thanks @PaulTaykalo and than you for updating the fixture to aid demoing and verifying this works 👍

**Comment by pepicrft** [comment]
@all-contributors add @PaulTaykalo for code

**Comment by PaulTaykalo** [comment]
@pepicrft Sure. I think that this is something that came from `main` branch.
Anyway, can't commit now until this fix https://github.com/tuist/tuist/pull/5426/files#diff-716e9196614e4df70d21cfe5515e79b2695a42931ca929e2654c88e27c081245R4 will be merged in.

**Comment by PaulTaykalo** [comment]
Okay, I made the same fix as in `xcode15`. This should allow merging PR's in any order. Thanks

**Comment by pepicrft** [comment]
> Okay, I made the same fix as in xcode15. This should allow merging PR's in any order. Thanks

Weird. It ignores the change :/. I'll look into it on Monday

**Comment by PaulTaykalo** [comment]
Heh, it just I forgot to add `.swiftformat` and since it locally passed all the tests, I thought that work is done :)

**Pull request merged**